src/hotspot/share/cds/filemap.cpp

```diff
@@ -480,6 +480,25 @@ void FileMapInfo::copy_shared_path_table(ClassLoaderData* loader_data, TRAPS) {
   }
 }
 
+void FileMapInfo::clone_shared_path_table(TRAPS) {
+  Arguments::assert_is_dumping_archive();
+
+  ClassLoaderData* loader_data = ClassLoaderData::the_null_class_loader_data();
+  ClassPathEntry* jrt = ClassLoader::get_jrt_entry();
+
+  assert(jrt != NULL,
+         "No modular java runtime image present when allocating the CDS classpath entry table");
+
+  // After dynamic dump, _saved_shared_path_table is corrupt, cannot be used again.
+  // Could not free the corrupt table. May have memory leak but it should be OK since
+  // it is not a big leak. This will fail:
+  // if (_saved_shared_path_table.size() > 0) {
+  //   MetadataFactory::free_array<u8>(loader_data, _saved_shared_path_table->_table);
+  // }
```

Comment:
In `FileMapInfo::copy_shared_path_table()`:

```
  Array<u8>* array = MetadataFactory::new_array<u8>(loader_data, (int)bytes, CHECK);
  _saved_shared_path_table = SharedPathTable(array, _shared_path_table.size());
```

You can remember the `array` pointer, e.g.

```
  _saved_shared_path_table_array = array;
```


and then call `MetadataFactory::free_array<u8>(_saved_shared_path_table_array)` in this function.

Reply:
Thanks. Remembering for later deletion is OK.